ReactWindows/ReactNative/ReactApplicationDelegate.cs

```diff
@@ -71,6 +80,19 @@ public void OnActivated(IActivatedEventArgs args)
             }
         }
 
+        /// <summary>
+        /// Creates the root view for the React app.
+        /// </summary>
+        /// <param name="arguments">The native app launch arguments.</param>
+        /// <returns>The root view.</returns>
+        public ReactRootView OnCreate(string arguments)
+        {
+            var host = _reactApplication.Host;
+            host.OnResume(_application.Exit);
+            host.ApplyArguments(arguments);
+            return host.OnCreate(LaunchOptions);
```

Comment:
This might work - but it strays a bit from the Android API if I'm not mistaken (https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/ReactActivityDelegate.java#L90)

The delegate passes its launch options to the RootView view `startReactApplication`.

Should the rnw delegate do the same?

Reply:
Good eye @jpshelley - `host.OnCreate(LaunchOptions)` does indeed pass the initial props to the `startReactApplication` call. The missing component here (that I called out above) is that we do not yet include an equivalent of the `createRootView` virtual from Android. Once we do that, we'll have to abandon the extension method from `ReactNativeHostExtensions` and make the calls here directly in `ReactApplicationDelegate` to account for the potentially overridden `createRootView`.

I suspect a PR for that will come soon as I'm investigating alternative options for touch handling, and overriding the root view may be the best way to support the new touch implementation (while simultaneously providing legacy support).

Reply:
Awesome sounds great - my background is in Android dev and not Windows so anything I can do to help I'll try and chip in. If you need eyes on PRs feel free to mention me and I'll try go respond faster then I did this last time 😆